### PR TITLE
fix(smart-contract): xdeployer setup eth conversion proxy

### PR DESCRIPTION
**Pull request by olivier7delf**

### CODE REVIEW

File: packages/smart-contracts/scripts-create2/contract-setup/setupERC20SwapToConversion.ts [hunk 1]
```diff
@@ -21,31 +21,25 @@ export const setupERC20SwapToConversion = async (
...
-        provider = utils.getDefaultProvider(network);
-      }
-      const wallet = new hre.ethers.Wallet(hre.config.xdeploy.signer, provider);
-      const signer = wallet.connect(provider);
-      const ERC20SwapToConversionConnected = await ERC20SwapToConversionContract.connect(signer);
-      const adminNonce = await signer.getTransactionCount();
-      const gasPrice = await provider.getGasPrice();
+  for (const network of hre.config.xdeploy.networks) {
+    await Promise.all(
+      [network].map(async (network) => {
```
Comment (leoslr) [inline packages/smart-contracts/scripts-create2/contract-setup/setupERC20SwapToConversion.ts:26]: Why using a mapping here and not directly the `network` value ?

File: packages/smart-contracts/scripts-create2/contract-setup/setupETHConversionProxy.ts [hunk 2]
```diff
@@ -15,33 +15,32 @@ export const setupETHConversionProxy = async (
...
-        provider = utils.getDefaultProvider(network);
-      }
-      const wallet = new hre.ethers.Wallet(hre.config.xdeploy.signer, provider);
-      const signer = wallet.connect(provider);
-      const EthConversionProxyConnected = await EthConversionProxyContract.connect(signer);
-      const adminNonce = await signer.getTransactionCount();
-      const gasPrice = await provider.getGasPrice();
+  for (const network of hre.config.xdeploy.networks) {
+    await Promise.all(
+      [network].map(async (network) => {
```
Comment (leoslr) [inline packages/smart-contracts/scripts-create2/contract-setup/setupETHConversionProxy.ts:22]: Same here

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 3]
```diff
@@ -75,6 +122,7 @@ export const updatePaymentErc20FeeProxy = async (
   }
 };
 
+/** legacy from BatchPayment */
```
Comment (olivier7delf) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:125]: Useless functions are gonna be deleted in the batch conversion deployment PR

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 4]
```diff
@@ -54,11 +65,47 @@ export const updateBatchPaymentFees = async (contract: any, gasPrice: BigNumber)
...
+      gasPrice: gasPrice,
+    });
+    await tx.wait();
+    console.log(
+      `${proxyType} conversion proxy: the current address ${currentAddress} has been replaced by: ${proxyAddress}`,
+    );
+  }
+};
+
+/** legacy from BatchPayment */
```
Comment (olivier7delf) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:108]: Useless functions are gonna be deleted in the batch conversion deployment PR

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 5]
```diff
@@ -10,6 +10,14 @@ export const REQUEST_SWAP_FEES = 5;
 // Batch Fees: .3%
 export const BATCH_FEE = 3;
 
+/**
+ * It update the chainlink address used by the contract.
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:14]:
```suggestion
 * Updates the chainlink address used by the contract.
```

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 6]
```diff
@@ -54,11 +65,47 @@ export const updateBatchPaymentFees = async (contract: any, gasPrice: BigNumber)
   if (!currentFees.eq(BATCH_FEE)) {
     const tx = await contract.setBatchFee(BATCH_FEE, { gasPrice: gasPrice });
     await tx.wait(1);
-    // Log is useful to have a direct view on what is being updated
     console.log(`currentFees: ${currentFees.toString()}, new fees: ${BATCH_FEE}`);
   }
 };
 
+/**
+ * Update the address of a Native or ERC20 fee proxy stored within a Native or ERC20 fee conversion contract
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:73]:
```suggestion
 * Updates the address of a Native or ERC20 fee proxy stored within a Native or ERC20 fee conversion contract
```

**Review by alexandre-abrioux — APPROVED** [review]

**Comment by alexandre-abrioux** [comment]
@leoslr If you take a look into // deployments, make sure to add the current chain in all `console.log/warn/error` (like add a `[chain]` prefix for instance) to know which one corresponds to which chain.

**Pull request merged**